node/tests/GlideClusterClient.test.ts

```diff
@@ -1966,4 +2002,390 @@ describe("GlideClusterClient", () => {
+                    } else {
+                        throw new Error(
+                            "Unexpected response format from INFO command",
+                        );
+                    }
+                } finally {
+                    await client_for_config_set?.close();
+                    await client_for_testing_az?.close();
+                }
+            },
+        );
+    });
+    describe("GlideClusterClient - AZAffinity with Non-existing AZ", () => {
+        it.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+            "should route commands to a replica when AZ does not exist using protocol %p",
+            async (protocol) => {
+                const GET_CALLS = 4;
+                const get_cmdstat = `cmdstat_get:calls=${GET_CALLS}`;
+                let client_for_testing_az;
+
+                try {
+                    if (azCluster.checkIfServerVersionLessThan("8.0.0")) {
+                        console.log(
+                            "Skipping test: requires Valkey 8.0.0 or higher",
+                        );
+                        return;
+                    }
+
+                    client_for_testing_az =
+                        await GlideClusterClient.createClient(
+                            getClientConfigurationOption(
+                                azCluster.getAddresses(),
+                                protocol,
+                                {
+                                    readFrom: "AZAffinity",
+                                    clientAz: "non-existing-az",
+                                    requestTimeout: 2000,
+                                },
+                            ),
+                        );
+
+                    await client_for_testing_az.customCommand([
+                        "CONFIG",
+                        "RESETSTAT",
+                    ]);
+
+                    for (let i = 0; i < GET_CALLS; i++) {
+                        await client_for_testing_az.get("foo");
+                    }
+
+                    const info_result =
+                        await client_for_testing_az.customCommand(
+                            ["INFO", "COMMANDSTATS"],
+                            { route: "allNodes" },
+                        );
+
+                    if (
+                        typeof info_result === "object" &&
+                        info_result !== null
+                    ) {
+                        const values = Object.values(info_result);
+                        const matching_entries_count = values.filter(
+                            (value) => {
+                                if (
+                                    !value ||
+                                    typeof value !== "object" ||
+                                    !("value" in value)
+                                )
+                                    return false;
+                                return value.value.includes(get_cmdstat);
```

Comment:
this test will fail. The AZ strategy searching foe replica in the same az, if not exits it will other replica in round robin. so in this case every replica will run the get command once